src/main/java/com/durov/maks/cinema/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,43 @@
+package com.durov.maks.cinema.controller;
+
+import com.durov.maks.cinema.model.dto.shoppingcart.ShoppingCartResponseDto;
+import com.durov.maks.cinema.model.mapper.ShoppingCartMapper;
+import com.durov.maks.cinema.service.MovieSessionService;
+import com.durov.maks.cinema.service.ShoppingCartService;
+import com.durov.maks.cinema.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final MovieSessionService movieSessionService;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+
+    public ShoppingCartController(MovieSessionService movieSessionService,
+                                  ShoppingCartService shoppingCartService,
+                                  UserService userService,
+                                  ShoppingCartMapper shoppingCartMapper) {
+        this.movieSessionService = movieSessionService;
+        this.shoppingCartMapper = shoppingCartMapper;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
+        return shoppingCartMapper.getShoppingCartResponseDto(shoppingCartService
+                .getByUser(userService.getUserById(userId)));
+    }
+
+    @PostMapping("/add-movie-session")
```

Comment:
we should not create a new resource for post method
```suggestion
    @PostMapping
```

Reply:
got it